### PR TITLE
[stable/prometheus-operator] Allow WAL compression + upgrade prometheus

### PR DESCRIPTION
**Pull request by elliotfehr**
#### What this PR does / why we need it:
- Upgrade default prometheus to latest release (2.12.0)
- Allow setting walCompression on the prometheus CRD

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Review by jspoth — APPROVED** [review]

File: stable/prometheus-operator/ci/test-values.yaml [hunk 1]
```diff
@@ -1395,6 +1395,14 @@ prometheus:
     ##
     retention: 10d
 
+    ## Maximum size of metrics
+    ##
+    retentionSize: ""
+
+    ## Enable compression of the write-ahead log using Snappy.
+    ##
+    walCompression: false
```
Comment (vsliouniaev) [inline stable/prometheus-operator/ci/test-values.yaml:1404]: This feature requires PrometheusOperator 0.32.0 https://github.com/coreos/prometheus-operator/releases/tag/v0.32.0

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/ci/test-values.yaml (hunk 1 above)
Comment (elliotfehr) [reply stable/prometheus-operator/ci/test-values.yaml:1404]: Fixed with https://github.com/helm/charts/pull/16635/commits/391433c7b167f2d95c0346c15b74a96d023aa8bb

**Pull request merged**